packages/issuer/lib/functions/CredentialOfferUtils.ts

```diff
@@ -0,0 +1,38 @@
+import { CredentialOfferPayload, CredentialOfferPayloadV1_0_11, encodeJsonAsURI, IssuerMetadata } from '@sphereon/openid4vci-common'
+import { v4 as uuidv4 } from 'uuid'
+
+export function createCredentialOfferURI(
+  issuerMetadata?: IssuerMetadata,
+  // todo: probably it's wise to create another builder for CredentialOfferPayload that will generate different kinds of CredentialOfferPayload
+  opts?: { state?: string; credentialOffer?: CredentialOfferPayload; preAuthorizedCode?: string; userPinRequired?: boolean }
```

Comment:
As discussed add an optional credentialOfferURI string value to opts. If present it will be used

Also add an optional scheme string to opts, replacing openid-credential-offer if present

Reply:
added the `credentialOfferURI` as discussed